asmcomp/cmmgen.ml

```diff
@@ -832,9 +832,7 @@ and transl_prim_1 env p arg dbg =
        dbg)
   | Pdls_get ->
       Cop(Cdls_get, [transl env arg], dbg)
-  | Patomic_load {immediate_or_pointer = Immediate} ->
-      Cop(mk_load_atomic Word_int, [transl env arg], dbg)
-  | Patomic_load {immediate_or_pointer = Pointer} ->
+  | Patomic_load ->
       Cop(mk_load_atomic Word_val, [transl env arg], dbg)
```

Comment:
This is the only use of the `mk_load_atomic` function. It may be better to remove it and inline the definition here.

Reply:
In general I prefer keeping things in Cmm_helpers when possible. Not only will it help when upstreaming Flambda2, but I think Cmmgen is easier to read and understand when it's not full of low-level encoding details.

Reply:
Sounds reasonable. In that case, it may be best to drop `memory_chunk` argument for `mk_load_atomic` as the only use fixes it to `Word_val`.

Reply:
Right, I will apply this extra simplification.

Reply:
Thinking more about this -- I was about to make the change -- I am not sure this is a good idea anymore, because it makes the logic in cmm_helpers.ml less regular. Currently we have:

```ocaml
(* Loads *)

let mk_load_immut memory_chunk =
  Cload {memory_chunk; mutability=Immutable; is_atomic=false}

let mk_load_mut memory_chunk =
  Cload {memory_chunk; mutability=Mutable; is_atomic=false}

let mk_load_atomic memory_chunk =
  Cload {memory_chunk; mutability=Mutable; is_atomic=true}
```

and I like this regularity. So I think that I will not change this for now.

Reply:
I’m fine with leaving it as is.